commit b6095cbd7841fff310c1ad135ae2c7da14df9da4
Author: Christophe JAILLET <christophe.jaillet@wanadoo.fr>
Date:   Mon Feb 4 08:54:20 2019 -0800

    xfs: Fix error code in 'xfs_ioc_getbmap()'
    
    commit 132bf6723749f7219c399831eeb286dbbb985429 upstream.
    
    In this function, once 'buf' has been allocated, we unconditionally
    return 0.
    However, 'error' is set to some error codes in several error handling
    paths.
    Before commit 232b51948b99 ("xfs: simplify the xfs_getbmap interface")
    this was not an issue because all error paths were returning directly,
    but now that some cleanup at the end may be needed, we must propagate the
    error code.
    
    Fixes: 232b51948b99 ("xfs: simplify the xfs_getbmap interface")
    Signed-off-by: Christophe JAILLET <christophe.jaillet@wanadoo.fr>
    Reviewed-by: Christoph Hellwig <hch@lst.de>
    Reviewed-by: Darrick J. Wong <darrick.wong@oracle.com>
    Signed-off-by: Darrick J. Wong <darrick.wong@oracle.com>
    Signed-off-by: Luis Chamberlain <mcgrof@kernel.org>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/fs/xfs/xfs_ioctl.c b/fs/xfs/xfs_ioctl.c
index 0ef5ece5634c..bad90479ade2 100644
--- a/fs/xfs/xfs_ioctl.c
+++ b/fs/xfs/xfs_ioctl.c
@@ -1616,7 +1616,7 @@ xfs_ioc_getbmap(
 	error = 0;
 out_free_buf:
 	kmem_free(buf);
-	return 0;
+	return error;
 }
 
 struct getfsmap_info {
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1241_b6095cbd7841fff310c1ad135ae2c7da14df9da4.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1241_b6095cbd7841fff310c1ad135ae2c7da14df9da4.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
